bin/validate_dn_format.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import os
+import re
+
+regex = re.compile(r"^(/(?:DC|OU|C|ST|L|O)=[^/=]+)*(/CN=[^/=].+)$")
+regexCN = re.compile(r"(?<=CN=).*")
+
+errorLog = []
+
+def validate(dir):
+    for root, subdirs, files in os.walk(dir):
+        for file in files:
+            if file.endswith('.lsc'):
+                filePath = os.path.join(root, file)
+                fileName = file[:-len('.lsc')]
+                with open(filePath, 'r') as f:  # reads .lsc files line by line
+                    lines = f.read().splitlines()
+                    if len(lines) != 2:
+                        errorLog.append("Error in \"" + filePath + "\" .lsc file should contain exactly 2 DNs")
+                    subjectCN = regexCN.search(lines[0]).group()  # reads subject DN and gets the CN
+                    for line in lines:
+                        match = regex.search(line)
+                        if not match:
+                            errorLog.append("Error in \"" + filePath + "\" at " + line)
+                        else:
+                            if line == lines[0]:
+                                if fileName not in subjectCN:
+                                    errorLog.append("Error in \"" + filePath + "\" at " + line)
+                                continue
+                            else:
+                                continue
```

Comment:
This `subjectCN` check is specific to `lines[0]`, so it does not belong inside the `for` loop.

```suggestion
                    if fileName not in subjectCN:
                        errorLog.append(f'Error in "{filePath}" at {lines[0]}')
```

(Separately, i don't know if that convention (that the hostname matches the CN, optionally after a service prefix) is actually a requirement or not.)

Reply:
> (Separately, i don't know if that convention (that the hostname matches the CN, optionally after a service prefix) is actually a requirement or not.)

@RyanEweSong BrianB should know for sure so let's bring this up at tomorrow's stand up meeting

Reply:
It is one of the checks listed in the jira ticket.